pkg/runtime/setup.go

```diff
@@ -553,6 +544,33 @@ func (s *setup) uninstall(id strfmt.UUID) (rerr error) {
 		return errs.Wrap(err, "Could not unload artifact envdef")
 	}
 
+	// If this is a dynamically imported artifact, tell the ecosystem to remove/undeploy it.
+	if artifact, exists := s.depot.config.Cache[id]; exists && artifact.Namespace != "" {
+		if ecosys := filterEcosystemMatchingNamespace(s.ecosystems, artifact.Namespace); ecosys != nil {
+			installedFiles := []string{}
+			// Find record of our deployment
+			if deployments, ok := s.depot.config.Deployments[id]; ok {
+				deployments = sliceutils.Filter(deployments, func(d deployment) bool { return d.Path == s.path })
```

Comment:
Might need `fileutils.ResolveUniquePath()` here. Note I thought we had something like `fileutils.IsEqualPath()` but apparently not, could be worth adding.